src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/DeclarationOrder/context/defaultContextConfig.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+          "https://checkstyle.org/dtds/configuration_1_3.dtd">
+<module name="Checker">
+    <module name="TreeWalker">
+        <module name="DeclarationOrder"/>
+
+        <module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
+            <property name="file" value="${tp}/defaultContext.patch" />
+            <property name="strategy" value="context" />
+            <property name="neverSuppressedChecks" value="DeclarationOrderCheck" />
```

Comment:
Why do you put it under this group? Can we use `checkNamesForContextStrategyByTokenOrParentSet` option instead?

Reply:
```
Test.java:6:5: Constructor definition in wrong order.
```
its violation node is `CTOR_DEF [6:4]`, if we use `checkNamesForContextStrategyByTokenOrParentSet`, its parent node is `OBJBLOCK [3:18]`, it's ok. 

```
|--CTOR_DEF -> CTOR_DEF [6:4]
    |   |--MODIFIERS -> MODIFIERS [6:4]
    |   |--IDENT -> Test [6:4]
```

```
Test.java:7:5: Instance variable definition in wrong order.
Test.java:8:5: Instance variable definition in wrong order.
```
But these two violation nodes are MODIFIERS [7:4], not  VARIABLE_DEF [7:4],  if we use `checkNamesForContextStrategyByTokenOrParentSet`, also can not solve this problem.

```
|--VARIABLE_DEF -> VARIABLE_DEF [7:4]
    |   |--MODIFIERS -> MODIFIERS [7:4]
    |   |--TYPE -> TYPE [7:4]
```